// react-dnd-touch-backend/lib/OptionsReader.js
export class OptionsReader {
    args;
    context;
    constructor(args, context) {
        this.args = args;
        this.context = context;
    }
    get delay() {
        return this.args.delay ?? 0;
    }
    get scrollAngleRanges() {
        return this.args.scrollAngleRanges;
    }
    get getDropTargetElementsAtPoint() {
        return this.args.getDropTargetElementsAtPoint;
    }
    get ignoreContextMenu() {
        return this.args.ignoreContextMenu ?? false;
    }
    get enableHoverOutsideTarget() {
        return this.args.enableHoverOutsideTarget ?? false;
    }
    get enableKeyboardEvents() {
        return this.args.enableKeyboardEvents ?? false;
    }
    get enableMouseEvents() {
        return this.args.enableMouseEvents ?? false;
    }
    get enableTouchEvents() {
        return this.args.enableTouchEvents ?? true;
    }
    get touchSlop() {
        return this.args.touchSlop || 0;
    }
    get delayTouchStart() {
        return this.args?.delayTouchStart ?? this.args?.delay ?? 0;
    }
    get delayMouseStart() {
        return this.args?.delayMouseStart ?? this.args?.delay ?? 0;
    }
    get window() {
        if (this.context && this.context.window) {
            return this.context.window;
        }
        else if (typeof window !== 'undefined') {
            return window;
        }
        return undefined;
    }
    get document() {
        if (this.context?.document) {
            return this.context.document;
        }
        if (this.window) {
            return this.window.document;
        }
        return undefined;
    }
    get rootElement() {
        return this.args?.rootElement || this.document;
    }
}
